test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt

```diff
@@ -15,49 +15,82 @@
  */
 package com.epam.drill.test2code.coverage
 
+import com.epam.drill.plugins.test2code.common.api.*
+import com.github.luben.zstd.*
 import kotlinx.coroutines.*
-import mu.KotlinLogging
-import java.util.concurrent.Executors
-import kotlin.coroutines.CoroutineContext
+import kotlinx.serialization.protobuf.*
+import mu.*
+import java.util.*
+import java.util.concurrent.*
 
 interface CoverageSender {
-    fun setSendingHandler(handler: SendingHandler)
     fun startSendingCoverage()
     fun stopSendingCoverage()
 }
 
 class IntervalCoverageSender(
-    intervalMs: Long,
-    collectProbes: () -> Sequence<ExecDatum> = { emptySequence() }
+    private val intervalMs: Long,
+    private val coverageTransport: CoverageTransport,
+    private val inMemoryBuffer: InMemoryBuffer = InMemoryBuffer(),
+    private val collectProbes: () -> Sequence<ExecDatum> = { emptySequence() }
 ) : CoverageSender {
     private val logger = KotlinLogging.logger {}
-    private var sendingHandler: SendingHandler = {}
-
-    private val job = ProbeWorker.launch(start = CoroutineStart.LAZY) {
-        while (isActive) {
-            delay(intervalMs)
-            sendingHandler(collectProbes())
-        }
-    }
-
-    override fun setSendingHandler(handler: SendingHandler) {
-        sendingHandler = handler
-    }
+    private val scheduledThreadPool = Executors.newSingleThreadScheduledExecutor()
 
     override fun startSendingCoverage() {
-        job.start()
+        scheduledThreadPool.scheduleAtFixedRate(
+            Runnable { sendProbes(collectProbes()) },
+            intervalMs,
+            intervalMs,
+            TimeUnit.MILLISECONDS
+        )
         logger.debug { "Coverage sending job is started." }
     }
 
     override fun stopSendingCoverage() {
-        job.cancel()
+        scheduledThreadPool.shutdown()
         logger.debug { "Coverage sending job is stopped." }
     }
-}
 
-internal object ProbeWorker : CoroutineScope {
-    override val coroutineContext: CoroutineContext = run {
-        // TODO ProbeWorker thread count configure via env.variable?
-        Executors.newFixedThreadPool(2).asCoroutineDispatcher() + SupervisorJob()
+    /**
+     * Create a function which sends chunks of test coverage to the admin part of the plugin
+     * @return the function of sending test coverage
+     * @features Coverage data sending
+     */
+    private fun sendProbes(data: Sequence<ExecDatum>) {
+        val compressedStrings = data
+            .map {
+                ExecClassData(
+                    id = it.id,
+                    className = it.name,
```

Comment:
TODO - Remove className from ExecClassData (id is enough)
issue: js-agent does not handle id-s at the moment